zephyr/esp32/src/wifi/esp_wifi_adapter.c

```diff
@@ -37,6 +37,36 @@ LOG_MODULE_REGISTER(esp32_wifi_adapter, CONFIG_WIFI_LOG_LEVEL);
 
 ESP_EVENT_DEFINE_BASE(WIFI_EVENT);
 
+#ifdef CONFIG_ESP32_WIFI_HEAP_MEM_POOL
+
+#define TAG "wifi"
+
+/* ESP heap extension */
+extern unsigned z_mapped_end;
+extern unsigned _heap_sentry;
+
+static void *esp_wifi_heap_init_mem = &z_mapped_end;
+#define ESP_WIFI_HEAP_SIZE  ((uintptr_t)&_heap_sentry - (uintptr_t)&z_mapped_end)
+
+static struct k_heap esp_wifi_heap;
+
+static int esp_wifi_heap_init(void)
+{
+	ESP_EARLY_LOGI(TAG, "ESP Wi-Fi heap init at 0x%x size %d kB.\n",
+			esp_wifi_heap_init_mem, ESP_WIFI_HEAP_SIZE/1024);
+	k_heap_init(&esp_wifi_heap, esp_wifi_heap_init_mem, ESP_WIFI_HEAP_SIZE);
+
+#if defined(CONFIG_WIFI) && defined(CONFIG_BT)
```

Comment:
Can't we do this in build time?

Reply:
I don't think it is possible. At least it wouldn't be a straightforward solution.

Reply:
what about `BUILD_ASSERT()`?

Reply:
> BUILD_ASSERT()

the expression behind the `ESP_WIFI_HEAP_SIZE` is not constant, same situation.